feat(phone-number): add phone number (#133)
siemens/element: docs/components/forms-inputs/phone-number-input.md

@Component({
  imports: [SiPhoneNumberInputComponent, ...]
})
```
<!-- markdownlint-disable MD046-->
??? info "Required Packages"
    - [google-libphonenumber](https://www.npmjs.com/package/google-libphonenumber)
    - [@types/google-libphonenumber](https://www.npmjs.com/package/@types/google-libphonenumber)
    - [flag-icons](https://www.npmjs.com/package/flag-icons)

    The currently supported dependency versions can be found in the [package.json of the demo app](https://github.com/siemens/element/blob/main/package.json).
<!-- markdownlint-enable MD046-->

- Ensure to import the `node_modules/flag-icons/css/flag-icons.css` to the `angular.json` styles configuration as follows:

```ts
          "styles": [
              "node_modules/flag-icons/css/flag-icons.css",
              ...
          ]
```

- To pass a custom list of country for the user to choose, ensure to pass the [ISO2](https://en.wikipedia.org/wiki/ISO_3166-2) codes of the countries as an array to `supportedCountries` input.
- To display all the available countries, don't add `supportedCountries` input.
- To select a default country in the dropdown, pass the ISO2 code of the country to the `defaultCountry` input.

> - `validatePhoneNumber` input needs to be set to true, if you wish to enable validation service and show valid/invalid status of the form control.

Reviewer: ```suggestion
- Set `validatePhoneNumber` input to enable a validation service and show status information in the form control.
```

> - The placeholder text for the country search input can be customized through the `placeholderForSearch` input.

Reviewer: ```suggestion
- Use the `placeholderForSearch` input for a custom placeholder text of the country search.
```

> - The ARIA label for the country dropdown can be customized through the `selectCountryAriaLabel` input.

Reviewer: ```suggestion
- The `selectCountryAriaLabel` input sets the ARIA label for the country dropdown .
```

> - The ARIA label for the phone number input can be customized through the `phoneNumberAriaLabel` input.

Reviewer: ```suggestion
- The `phoneNumberAriaLabel` input sets the ARIA label for the phone number input.
```

> - The form control tagged to this phone number component will be assigned a validation error of `invalidPhoneNumber`, whenever `validatePhoneNumber` is set to `true` and an invalid phone number is entered.

Reviewer: ```suggestion
```

> - On any change of data, the phone number component will emit the current value of the component through the `valueChange` output.

Reviewer: ```suggestion
- All changes are emitted through the `valueChange` output.
```
